rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace.
+      responses:
+        200:
+          description: OK - Table Exists
+        401:
+          description: Unauthorized
+        404:
+          description: Not Found
+
+  /v1/tables/rename:
+    post:
+      tags:
+        - Catalog API
+      summary: Rename a table from its current name to a new name
+      description:
+        Rename a table from one identifier to another. It's valid to move a table
+        across namespaces, but the server implementation doesn't need to support it.
+      operationId: renameTable
+      requestBody:
+        description: Current table identifier to rename and new table identifier to rename to
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RenameTableRequest'
+        required: true
+      responses:
+        200:
+          description: OK
+        401:
+          description: Unauthorized
+        404:
+          description:
+            Not Found
+            - NoSuchTableException, Table to rename does not exist
+            - NoSuchNamespaceException, The target namespace of the new table identifier does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                TableToRenameDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceToRenameToDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        406:
+          description: Not Acceptable - Unsupported Operation
+        409:
+          description: Conflict - The target table identifier to rename to already exists
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              example:
+                $ref: '#/components/examples/TableAlreadyExistsError'
+
+components:
+  #######################################################
+  # Common Parameter Definitions Used In Several Routes #
+  #######################################################
+  parameters:
+    namespace:
+      name: namespace
+      in: path
+      required: true
+      description:
+        A namespace identifier as a single string.
+        Multipart namespace parts should be separated by the null byte, %00.
+      schema:
+        type: string
+      examples:
+        singlepart_namespace:
+          value: "accounting"
+        multipart_namespace:
+          value: "accounting%00tax"
+
+    table:
+      name: table
+      in: path
+      description: A table name
+      required: true
+      schema:
+        type: string
+      example: "sales"
+
+    parent:
+      name: parent
+      in: query
+      description:
+        Optional parent namespace under which to list namespaces.
```

Comment:
If this is generic, then we should just say it is an optional parent namespace, without necessarily stating what its purpose is ("under which to list namespaces").

Reply:
That's really the only place it's used. I actually considered just leaving it as a parameter in the one spot it's used. but since I didn't, I'll remove that `under which to list namespaces` part.

Reply:
Actually, since it has the description of the behavior of what would happen if it's not included, I think it should go under the parameters section of the single route that uses it. Let me think about it and see if there's a good place to document that behavior and leave this generic down here.

Reply:
Ultimately, this isn't a generic parameter at the moment, so I moved it back to the one route where it is used. If we find need of it to be a generic parameter in the future, then we can deal with that then.